src/public/common/search/Option.js

```diff
@@ -1,115 +1,87 @@
-        <TargetIcon />
-        {data.approvedSymbol}
-      </Typography>{' '}
-      <Typography>{data.approvedName}</Typography>
-      {data.proteinAnnotations ? (
-        <Typography variant="caption" noWrap display="block">
-          {data.proteinAnnotations.functions[0]}
-        </Typography>
-      ) : null}
-    </>
-  );
-};
+const TopHitDisease = ({ data }) => (
+  <>
+    <Typography variant="h4" color="primary">
+      <DiseaseIcon /> {data.name}
+    </Typography>
+    <Typography variant="caption" display="block" noWrap>
+      {data.description}
+    </Typography>
+  </>
+);
 
-const DiseaseTopHit = ({ data }) => {
-  return (
-    <>
-      <Typography variant="h4" color="primary">
-        <DiseaseIcon />
-        {data.name}
+const TopHitDrug = ({ data }) => (
+  <>
+    <Typography variant="h4" color="primary">
+      <DrugIcon /> {data.name}
+    </Typography>
+    {data.mechanismsOfAction ? (
+      <Typography variant="caption" display="block" noWrap>
+        {data.mechanismsOfAction.rows
+          .map(row => row.mechanismOfAction)
+          .join(', ')}
       </Typography>
-      <Typography variant="caption" noWrap display="block">
-        {data.description}
-      </Typography>
-    </>
-  );
-};
+    ) : null}
+  </>
+);
 
-const DrugTopHit = ({ data }) => {
-  return (
-    <>
-      <Typography variant="h4" color="primary">
-        <DrugIcon />
-        {data.name}
+const TopHitTarget = ({ data }) => (
+  <>
+    <Typography variant="h4" color="primary">
+      <TargetIcon /> {data.approvedSymbol}
+    </Typography>{' '}
+    <Typography display="block" noWrap>
+      {data.approvedName}
+    </Typography>
+    {data.proteinAnnotations ? (
+      <Typography variant="caption" display="block" noWrap>
+        {data.proteinAnnotations.functions[0]}
       </Typography>
-      {data.mechanismsOfAction ? (
-        <Typography variant="caption" noWrap display="block">
-          {data.mechanismsOfAction.rows
-            .map(row => row.mechanismOfAction)
-            .join(', ')}
-        </Typography>
-      ) : null}
-    </>
-  );
-};
+    ) : null}
+  </>
+);
 
-const TopHit = ({ data }) => {
-  return data.__typename === 'Target' ? (
-    <TargetTopHit data={data} />
-  ) : data.__typename === 'Disease' ? (
-    <DiseaseTopHit data={data} />
-  ) : (
-    <DrugTopHit data={data} />
-  );
-};
+const Disease = ({ data }) => (
+  <Typography variant="subtitle2" display="inline">
+    {data.name}
+  </Typography>
+);
+
+const Drug = ({ data }) => (
+  <Typography variant="subtitle2" display="inline">
+    {data.name}
+  </Typography>
+);
+
+const Target = ({ data }) => (
+  <>
+    <Typography variant="subtitle2" display="inline">
+      {data.approvedSymbol}
+    </Typography>{' '}
+    <Typography variant="caption" color="textSecondary" display="inline">
+      {data.approvedName}
+    </Typography>
+  </>
+);
+
+const Option = ({ data }) => {
+  const optionTypes = {
```

Comment:
Please move this object outside of the component. It does not depend on either props or state and we don't want to create this object on every render.